### PR TITLE
[prometheus] Allow Volume expansion for pvc's

### LINKED ISSUE #338: [prometheus] Allow volume expansion
Describe *in detail* the feature/behavior/change you'd like to see.

Please allow volume expansion in the pvc storage class

Ref : https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/server/pvc.yaml

### PR DESCRIPTION
**Pull request by bseenu**
#### What this PR does / why we need it:
Allow Volume expansion for pvc's for alertmanager and prometheus

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #338

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x ] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus/templates/alertmanager/pvc.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ spec:
 {{- if .Values.alertmanager.persistentVolume.volumeBindingMode }}
   volumeBindingModeName: "{{ .Values.alertmanager.persistentVolume.volumeBindingMode }}"
 {{- end }}
+  allowVolumeExpansion: {{ .Values.alertmanager.persistentVolume.allowVolumeExpansion | default "false" | quote }}
```
Comment (timurista) [inline charts/prometheus/templates/alertmanager/pvc.yaml:28]: @bseenu  does this work on a pvc I though only on a storage class.
You can only expand a PVC if its storage class's allowVolumeExpansion field is set to true.
```
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  name: gluster-vol-default
provisioner: kubernetes.io/glusterfs
parameters:
  resturl: "http://192.168.10.100:8080"
  restuser: ""
  secretNamespace: ""
  secretName: ""
allowVolumeExpansion: true

Comment (Xtigyro) [inline charts/prometheus/templates/alertmanager/pvc.yaml:28]: @bseenu I believe this is down to the Storage Class config. Correct me please if I'm mistaken.

**Review by Xtigyro — CHANGES_REQUESTED** [review]

File: charts/prometheus/templates/alertmanager/pvc.yaml (hunk 1 above)
Comment (bseenu) [reply charts/prometheus/templates/alertmanager/pvc.yaml:28]: @timurista @Xtigyro Yes this is storage class config, but in the values this is exposed under persistentVolume, please check this https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/values.yaml#L213

Comment (bseenu) [reply charts/prometheus/templates/alertmanager/pvc.yaml:28]: @timurista @Xtigyro sorry my bad i got confused with storage class available in pvc template, yes this should be a seperate template for storage class

**Comment by bseenu** [comment]
This actually brings up one more bug , there is an attribute defined `volumeBindingModeName` which is actually defined for storageclass but included in the pvc template : https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus/templates/server/pvc.yaml#L27, which is not going to work

```
bash-3.2$ helm repo ls
NAME                	URL
kiali               	https://kiali.org/helm-charts
stable              	https://charts.helm.sh/stable
prometheus-community	https://prometheus-community.github.io/helm-charts
bash-3.2$ helm upgrade --install   prometheus --set  server.persistentVolume.volumeBindingMode="gp2" --set  server.persistentVolume.volumeBindingMode="WaitForFirstConsumer" prometheus-community/prometheus
Release "prometheus" does not exist. Installing it now.
Error: unable to build kubernetes objects from release manifest: error validating "": error validating data: ValidationError(PersistentVolumeClaim.spec): unknown field "volumeBindingModeName" in io.k8s.api.core.v1.PersistentVolumeClaimSpec
```

And volumeBindingMode is actually defined in storageclass and not persistent volume claim

Ref: https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#persistentvolumeclaimspec-v1-core
https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#storageclass-v1-storage-k8s-io

**Comment by Xtigyro** [comment]
Let's close this, guys. Thank you for your time.

**Pull request closed without merging**